src/binary/merkle_tree.rs

```diff
@@ -239,41 +255,49 @@ where
             .iter()
             .unzip();
 
-        let mut current_head = None;
-        let peaks = &peaks.as_slice()[1..]; // Omit the root.
-        for peak in peaks.iter() {
-            let key = peak.in_order_index();
-            let node = self
-                .storage
-                .get(&key)?
-                .ok_or(MerkleTreeError::LoadError(key))?
-                .into_owned()
-                .into();
-            let next = Box::new(Subtree::<Node>::new(node, current_head));
-            current_head = Some(next);
-        }
-
-        self.head = current_head;
+        peaks
+    }
 
-        Ok(())
+    fn root_position(&self) -> Position {
+        self.peak_positions()[0]
```

Comment:
Is there a way to do without `unzip`'ing the entire iterator if just the first value is needed with just a simple call to `take`?

Reply:
Yes, good point. I refactored these two functions to invert the dependencies: I calculate the root position independently inside `root_position`, and call that from `peak_positions`.